libs/components/src/lib/text-field/text-field.scss

```
@@ -44,6 +44,25 @@ $low-ink-color: --_low-ink-color;
 		#{variables.$text-field-icon-size}: calc(
 			#{size.$vvd-size-normal} / 2
 		); // default of 20px
+		#{variables.$text-field-block-size}: #{size.$vvd-size-normal};
+
+		display: inline-grid;
+		width: 100%;
+		gap: 4px;
+		grid-template-columns: min-content 1fr max-content;
+	}
+
+	&.size-condensed {
+		#{variables.$text-field-gutter-start}: calc(
+			#{size.$vvd-size-condensed} / 2.5
+		); // default of 16px
+		#{variables.$text-field-gutter-end}: calc(
+			#{size.$vvd-size-condensed} / 2.5
+		); // default of 16px
+		#{variables.$text-field-icon-size}: calc(
+			#{size.$vvd-size-condensed} / 2
+		); // default of 20px
```

the values that are in the comments are wrong.
 They actually defaults to 12.8px which is bad. I change the calculation to give the value of 12px

for the last one - the font size is 16px. Is this what needed?
```suggestion
		#{variables.$text-field-gutter-start}: calc(
			#{size.$vvd-size-super-condensed} / 2
		); // default of 12px
		#{variables.$text-field-gutter-end}: calc(
			#{size.$vvd-size-super-condensed} / 2
		); // default of 12px
		#{variables.$text-field-icon-size}: calc(
			#{size.$vvd-size-condensed} / 2
		); // default of 16px
```

---

we actually keep the convention of using :not(XXX) and &.XXX
so you actually need to add the code line this:
```
& {
		display: inline-grid;
		width: 100%;
		gap: 4px;
		grid-template-columns: min-content 1fr max-content;
	}

	&:not(.size-condensed ) {
		#{variables.$text-field-gutter-start}: calc(
			#{size.$vvd-size-normal} / 2.5
		); // default of 16px
		#{variables.$text-field-gutter-end}: calc(
			#{size.$vvd-size-normal} / 2.5
		); // default of 16px
		#{variables.$text-field-icon-size}: calc(
			#{size.$vvd-size-normal} / 2
		); // default of 20px
	}

	&.size-condensed  {
		#{variables.$text-field-gutter-start}: calc(
			#{size.$vvd-size-super-condensed} / 2
		); // default of 12px
		#{variables.$text-field-gutter-end}: calc(
			#{size.$vvd-size-super-condensed} / 2
		); // default of 12px
		#{variables.$text-field-icon-size}: calc(
			#{size.$vvd-size-condensed} / 2
		); // default of 16px
	}
```